src/compute/src/render/top_k.rs

```diff
@@ -215,6 +215,12 @@ where
                 for log_modulus in
                     [60, 56, 52, 48, 44, 40, 36, 32, 28, 24, 20, 16, 12, 8, 4u64].iter()
                 {
+                    // Avoid building stages larger than the expected group size, if we were given
+                    // that hint.
+                    if expected_group_size.map_or(false, |g| g < 1 << log_modulus) {
```

Comment:
> I suppose that an expected_group_size of 256 would lead to one bucket (16) above, but to two buckets (256 and 16) in the top-k variant?

That's true.  I'll change it to `<=` in the top-k case.

> Perhaps we can abstract a helper function that we can reuse to compute the buckets for BucketedPlan and also for the BasicTopKPlan? Then BasicTopKPlan could be given buckets when lowering from MIR to LIR instead of having them computed here. This alternative would also ensure that we set the same default expected_group_size for both types of plans.

That's reasonable, I'll give it a shot.

Reply:
Even if so, performance-wise that aspect is dwarfed by everything else going on.  In my same scratch instance scenario, the query times are largely the same, though the addition of the new default means that without the option, the same query runs much faster.  Using `EXPECTED GROUP SIZE = 18446744073709551615` to simulate the original behavior performs as expected.

Reply:
Perhaps we can leave the flexibility in for now, and revisit that after we work on https://github.com/MaterializeInc/materialize/issues/17135? That way, we can have some freedom on how to structure the parameter sweep there.

Reply:
I'm fine with this.